src/encode.cpp

```diff
@@ -71,6 +71,43 @@ namespace {
       return nullopt;
     }
   }
+
+  void encodeShiftAmountBound(State &st, mlir::Operation* op) {
+    const auto arg = op->getOperand(0);
+    const auto argTy = arg.getType();
+    const auto amnt = op->getOperand(1);
+    const auto amntTy = amnt.getType();
```

Comment:
I think variable `arg` is not necessary because (1) UB depends on the shift amount only, and (2) amntTy == argTy. What do you think?

Reply:
Then let's leave `smart_assert` to check whether types are equal.